### PR TITLE
Night Vision Goggles

### PR DESCRIPTION
**Pull request by Diegoflores31**
## About The Pull Request

**Nigth Vision Goggles:**
-Added new night vision goggles available for SLs and RTOs on their respective squad vendors . this item can be attached into your helmet to grant you the ability to see in dark enviroments with a green filter and some visual noise . 

-The night vision goggles have a limited duration of aproximatelly 5 minutes after which you will have to manually reload them using a battery to keep them working.

-xenomorphs slashing the head of the users will eventually break the NVGs making them unusable untill someone capable can repair them. reducing its durability and duration by half

-For special operations this item can be bougth in Req in a crate of 3 units

-Cannot be used with binoculars / scopes

**Tactical Night Vision Goggles:**

-Unlimited battery and durability variant of the traditional NVG units issued to MARSOC operators 

-This will replace the chunky marsoc NVGs with the new NVG system

## Why It's Good For The Game

It will provide more gameplay options to RTO and Squad Leaders making this jobs feel a little more special and granting them with exclusive gear to perform their duties while also balancing the way NVG works

## Changelog

a

:cl:
add: Night vision goggles
add: Tactical night vision goggles
add: NVG crate on REQ
/:cl:

### CODE REVIEW

**Review by Agameofscones — APPROVED** [review]
Map and icon changes are 👌

File: code/_onclick/click.dm [hunk 1]
```diff
@@ -281,6 +281,7 @@
 
 
 /client/proc/change_view(new_size, var/atom/source)
+	if(SEND_SIGNAL(mob, COMSIG_MOB_CHANGE_VIEW, new_size) & COMPONENT_OVERRIDE_VIEW) return TRUE
```
Comment (Watermelon914) [inline code/_onclick/click.dm:284]:
```suggestion
	if(SEND_SIGNAL(mob, COMSIG_MOB_CHANGE_VIEW, new_size) & COMPONENT_OVERRIDE_VIEW) 
		return TRUE
```

File: code/game/objects/items/props/helmetgarb.dm [hunk 2]
```diff
@@ -94,110 +116,342 @@
...
+			to_chat(user, "They appear to be in good shape.")
+		else if(nvg_health_procent > 50)
+			to_chat(user, "They are visibly damaged.")
+		else if(nvg_health_procent > 30)
+			to_chat(user, "It's unlikely they can sustain more damage.")
+		else if(nvg_health_procent >= 0)
+			to_chat(user, "They are falling apart.")
+
+	if (get_dist(user, src) <= 1 && (shape == NVG_SHAPE_FINE || shape == NVG_SHAPE_PATCHED))
+		to_chat(user, "A small gauge in the corner reads: Power: [round(100.0*nvg_charge/nvg_maxcharge) ]%.")
```
Comment (Watermelon914) [inline code/game/objects/items/props/helmetgarb.dm:233]: Separate these out into different procs.
Your do_afters should use the SECONDS defines as well

File: code/game/objects/items/props/helmetgarb.dm [hunk 3]
```diff
@@ -94,110 +116,342 @@
...
-	UnregisterSignal(attached_item, COMSIG_PARENT_QDELETING)
-	qdel(activation)
-	attached_item = null
-
-/obj/item/prop/helmetgarb/helmet_nvg/ui_action_click(var/mob/owner, var/obj/item/holder)
-	toggle_nods(owner)
-	activation.update_button_icon()
 
+/obj/item/prop/helmetgarb/helmet_nvg/proc/set_attached_mob(var/mob/A)
+	attached_mob = A
```
Comment (Watermelon914) [inline code/game/objects/items/props/helmetgarb.dm:241]: use a more detailed variable name for A

File: code/game/objects/items/props/helmetgarb.dm [hunk 4]
```diff
@@ -94,110 +116,342 @@
...
-	UnregisterSignal(attached_mob, COMSIG_HUMAN_POST_UPDATE_SIGHT)
-	update_sight(attached_mob)
-	attached_mob = null
+	if(new_size > 7) // cannot use binos with NVG
+		toggle_nods(M)
+
+/obj/item/prop/helmetgarb/helmet_nvg/proc/break_nvg(mob/living/carbon/human/user, list/slashdata, var/mob/living/carbon/Xenomorph/X) //xenos can break NVG if aim head
+	SIGNAL_HANDLER
+
+	if(check_zone(X.zone_selected) == "head" && user == attached_mob)
```
Comment (Watermelon914) [inline code/game/objects/items/props/helmetgarb.dm:427]: use a more detailed variable name for xenomorph X

**Review by Watermelon914 — CHANGES_REQUESTED** [review]

**Comment by usnpeepoo** [comment]
I recommend removing RTOs from purchasing this. Normally with only 4 players (SL) having Night vision  the impact on a round isn’t significant however on high pop there are 8 RTO slots available which immediately boosts the amount of marines having night vision to 16. The roles would just be taken to power game and abuse nightvisions and having it only set to SLs (who actually need love) would be preferable than having 16 night vision runners in a round.

**Comment by Diegoflores31** [comment]
changes applied

**Pull request closed without merging**